### ISSUE TITLE
Unable to linkify when link has multiple parameters

### ISSUE THREAD

**Issue by binarykitchen** [parshap/html-linkify#8]
```
https://videomail.io/?to=daniel@hello.co.nz&subject=Question blahblah 
```

ends up in a weird, broken link where the email address becomes another mailto link, but shouldn't.

**Comment by parshap** [comment]
This is the result of how the [regular expression being used][regex] to find links ([originally from here][gruber]) treats the `@` symbol. You may be able to fix this by changing the regex.

There is also another regex-based implementation used in [twitter-text][]. It's probably a better regex than the one currently used – maybe we can use it.

[regex]: https://github.com/parshap/html-linkify/blob/02bc652011592b43569d79e0dc92658a448783ec/index.js#L7
[gruber]: https://gist.github.com/gruber/8891611
[twitter-text]: https://github.com/twitter/twitter-text/blob/7f40a456a37d703d78030245684f849a915c39aa/js/twitter-text.js#L849

**Comment by binarykitchen** [comment]
@parshap I see. So, isn't this a bug within that regular expression https://github.com/parshap/html-linkify/blob/02bc652011592b43569d79e0dc92658a448783ec/index.js#L7 ?

Do you think you can fix that or have no time and prefer me to do a PR and fix that regex?

**Comment by parshap** [comment]
Yes, it is indeed a bug within that regular expression. I don't currently have the time to try and fix it, but would happily look at a PR.

**Comment by binarykitchen** [comment]
Alright, I ll think about it when I have time ...

**Comment by binarykitchen** [comment]
I am sorry, this is too hard and already tried one hour to fix that `rEmail` regex :(

**Comment by parshap** [comment]
Yes the regex is painful. I think the right approach is using a real lexer and parser.

**Comment by binarykitchen** [comment]
I'm afraid, I'll have to give this one a pass. Too hard for me right now. Hope someone else can do this one day?
